### PR TITLE
Added apmgoredisv8 module

### PR DESCRIPTION
**Pull request by idealism-xxm**
Add support for go-redis/redis/v8

### CODE REVIEW

File: module/apmgoredisv8/hook.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	"go.elastic.co/apm"
+)
+
+// goRedisApmHook is an implementation of redis.Hook that reports cmds as spans to Elastic APM.
+type goRedisApmHook struct{}
+
+// NewGoRedisApmHook returns a redis.Hook that reports cmds as spans to Elastic APM.
+func NewGoRedisApmHook() redis.Hook {
+	return &goRedisApmHook{}
+}
```
Comment (axw) [inline module/apmgoredisv8/hook.go:38]:
```suggestion
// hook is an implementation of redis.Hook that reports cmds as spans to Elastic APM.
type hook struct{}

// NewHook returns a redis.Hook that reports cmds as spans to Elastic APM.
func NewGoRedisApmHook() redis.Hook {
	return &hook{}
}
```

The package name provides the context of what type of hook it is, so having it in the function name stutters. Calling it "NewHook" is clear: `apmgoredisv8.NewHook()`.

File: module/apmgoredisv8/hook.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+var key = goRedisApmSpanKey{}
+
+func withSpan(ctx context.Context, span *apm.Span) context.Context {
+	return context.WithValue(ctx, key, span)
+}
+
+func spanFromContext(ctx context.Context) *apm.Span {
+	span, _ := ctx.Value(key).(*apm.Span)
+	return span
+}
```
Comment (axw) [inline module/apmgoredisv8/hook.go:96]:
```suggestion
```

The context returned by [apm.StartSpan](https://pkg.go.dev/go.elastic.co/apm?tab=doc#StartSpan) contains the started span. You can use [apm.SpanFromContext](https://pkg.go.dev/go.elastic.co/apm?tab=doc#SpanFromContext) to retrieve it.

**Comment by idealism-xxm** [comment]
@axw All issues are fixed now. Please review it and let me know if I've missed anything.

**Comment by axw** [comment]
jenkins run the tests please

**Pull request merged**